File: dotnet/Allmark/RenderConsole/ConsoleHardBreakRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleHardBreakRenderer
{
    public static OutputRenderer Create()
    {
        return new OutputRenderer
        {
            Name = "hard_break",
            Render = Render,
        };
    }

    public static void Render(MarkdownNode node, RendererState state, bool? first = null, bool? last = null, bool? decode = true)
    {
        state.Output.Append('\n');
    }
}

File: zig/src/render-console/consoleHardBreakRenderer.zig
const std = @import("std");

const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const ConsoleRendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub const consoleHardBreakRenderer = Renderer{
    .name = "hard_break",
    .render = render,
};

pub fn render(node: *const MarkdownNode, state: *ConsoleRendererState, first: ?bool, last: ?bool, decode: ?bool) void {
    _ = node;
    _ = first;
    _ = last;
    _ = decode;

    state.output.append(state.allocator, '\n') catch unreachable;
}
